### PR TITLE
WIP: Documentation 

### PR DESCRIPTION
**Pull request by UARTman**
This is the branch containing my ongoing efforts to help document the library (https://github.com/audulus/rui/issues/10). Please do not merge them yet.

If you have any problems with the docs, please tell me and/or edit the branch.

### CODE REVIEW

File: src/views/canvas.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ where
     }
 }
 
-/// Canvas for GPU drawing with Vger. See https://github.com/audulus/vger-rs.
+/// Canvas for GPU drawing with Vger. See <https://github.com/audulus/vger-rs>.
```
Comment (wtholliday) [inline src/views/canvas.rs:52]: Where does that warning come from?

Comment (UARTman) [reply src/views/canvas.rs:52]: When I ran `cargo doc`, it threw a warning (https://doc.rust-lang.org/rustdoc/lints.html#bare_urls). Essentially, as it was it wouldn't actually render into a hyperlink, just a URL text. Adding `<>` should make it a hyperlink.

Comment (UARTman) [reply src/views/canvas.rs:52]: I've just compared the version on docs.rs and my local docs build, and adding `<>` as suggested really did turn this into a hyperlink.

Before the change:
изображение

After:
изображение

Comment (UARTman) [reply src/views/canvas.rs:52]: The warning page I linked to explains why it doesn't show up unless you try and build the docs locally:

 > Note that, except for missing_docs, these lints are only available when running rustdoc, not rustc.

Comment (wtholliday) [reply src/views/canvas.rs:52]: Cool.. thanks for the info!

**Comment by wtholliday** [comment]
This seems like a good step. Why not just merge it now?

**Comment by maaku** [comment]
It's been two years. Should this be merged or closed?

**Comment by wtholliday** [comment]
@maaku probably rather stale at this point. I'll close it :)

**Pull request closed without merging**